src/librustc/hir/lowering/expr.rs

```diff
@@ -659,11 +659,23 @@ impl LoweringContext<'_> {
         // mut pinned => loop { ... }
         let pinned_arm = self.arm(pinned_pat, loop_expr);
 
+
+        // `match ::std::future::IntoFuture::into_future(<expr) { ... }`
+        let expr = P(self.lower_expr(expr));
+        let into_future_expr = {
+            let into_future_path =
+                &[sym::future, sym::IntoFuture, sym::into_future];
+            P(self.expr_call_std_path(
+                span,
```

Comment:
This won't work, I think... you need to mark the span as allowing the feature with https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/lowering/struct.LoweringContext.html#method.mark_span_with_reason.

Reply:
Oh cool! Got it, I think?